### PR TITLE
Add redfish provider:

### PR DESCRIPTION
**Pull request by jacobweinstock**
redfish provider supporting power state and power set

### CODE REVIEW

File: providers/redfish/redfish.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+func (r *Conn) hardoff(ctx context.Context) (ok bool, err error) {
+	service := r.conn.Service
+	ss, err := service.Systems()
+	if err != nil {
+		return false, err
+	}
+	for _, system := range ss {
+		if system.PowerState == rf.OnPowerState {
```
Comment (stmcginnis) [inline providers/redfish/redfish.go:167]: Is `OnPowerState` correct here? I think you only want to break if it's already off, right?

**Review by stmcginnis — COMMENTED** [review]
Looks good to me, just one possible copy/paste issue pointed out.

File: providers/redfish/redfish.go (hunk 1 above)
Comment (jacobweinstock) [reply providers/redfish/redfish.go:167]: yes, good catch. thank you!

**Comment by jacobweinstock** [comment]
required https://github.com/bmc-toolbox/bmclib/pull/215 first

File: providers/redfish/redfish.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+const (
+	// ProviderName for the provider implementation
+	ProviderName = "gofish"
+	// ProviderProtocol for the provider implementation
+	ProviderProtocol = "redfish"
+)
+
+var (
+	// Features implemented by ipmitool
```
Comment (joelrebel) [inline providers/redfish/redfish.go:25]: minor typo there - s/ipmitool/redfish/`

File: client.go [hunk 3]
```diff
@@ -75,6 +76,9 @@ func (c *Client) registerProviders() {
 	driverAsrockrack, _ := asrockrack.New(c.Auth.Host, c.Auth.User, c.Auth.Pass, c.Logger)
 	c.Registry.Register(asrockrack.ProviderName, asrockrack.ProviderProtocol, asrockrack.Features, nil, driverAsrockrack)
 
+	// register gofish provider
+	driverGoFish := &redfish.Conn{Host: c.Auth.Host, Port: c.Auth.Port, User: c.Auth.User, Pass: c.Auth.Pass, Log: c.Logger}
```
Comment (joelrebel) [inline client.go:80]: have to say, I'm quite excited about this driver :D

**Review by joelrebel — CHANGES_REQUESTED** [review]

File: providers/redfish/redfish.go (hunk 2 above)
Comment (jacobweinstock) [reply providers/redfish/redfish.go]: updated

**Review by joelrebel — APPROVED** [review]

**Pull request merged**